pkg/api/v20230701preview/openshiftcluster_validatestatic.go

```diff
@@ -228,6 +231,44 @@ func (sv openShiftClusterStaticValidator) validateNetworkProfile(path string, np
 			return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path+".outboundType", "The provided outboundType '%s' is invalid: cannot use UserDefinedRouting if either API Server Visibility or Ingress Visibility is public.", np.OutboundType)
 		}
 	}
+
+	if np.OutboundType == OutboundTypeUserDefinedRouting && np.LoadbalancerProfile != nil {
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path+".loadBalancerProfile", "The provided loadBalancerProfile is invalid: cannot use a loadBalancerProfile if outboundType is UserDefinedRouting.")
+	}
+
+	return nil
+}
+
+func (sv openShiftClusterStaticValidator) validateLoadBalancerProfile(path string, lbp *LoadbalancerProfile, isCreate bool) error {
+	if lbp != nil {
+		var isManagedOutboundIPCount = lbp.ManagedOutboundIPs != nil
+		var isOutboundIPs = lbp.OutboundIPs != nil
+		var isOutboundIPPrefixes = lbp.OutboundIPPrefixes != nil
+
+		if isManagedOutboundIPCount && !isOutboundIPPrefixes && !isOutboundIPs {
+			if !(lbp.ManagedOutboundIPs.Count > 0 && lbp.ManagedOutboundIPs.Count <= 20) {
+				return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path+".managedOutboundIps.count", "The provided managedOutboundIps.count %d is invalid: managedOutboundIps.count must be in the range of 1 to 20 (inclusive).", lbp.ManagedOutboundIPs.Count)
+			}
+		} else if isOutboundIPs && !isOutboundIPPrefixes && !isManagedOutboundIPCount {
+			return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path+".outboundIps", "The field outboundIps is not implemented at this time, please check back later.")
+		} else if isOutboundIPPrefixes && !isOutboundIPs && !isManagedOutboundIPCount {
+			return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path+".outboundIpPrefixes", "The field outboundIpPrefixes is not implemented at this time, please check back later.")
+		} else if !isManagedOutboundIPCount && !isOutboundIPPrefixes && !isOutboundIPs {
+			return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path, "The provided loadBalancerProfile is invalid: must specify one of managedOutboundIps, outboundIps, or outboundIpPrefixes.")
+		} else {
+			return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path, "The provided loadBalancerProfile is invalid: can only use one of managedOutboundIps, outboundIps, or outboundIpPrefixes at a time.")
+		}
+
+		if lbp.AllocatedOutboundPorts != nil {
+			return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path+".allocatedOutboundPorts", "The field allocatedOutboundPorts is not implemented at this time, please check back later.")
```

Comment:
What is the plan to implement this?

Reply:
This is will be implemented in phase 3, so after `managedOutboundIPs`, `outboundIPs` and `outboundIPPrefixes` are implemented.  There will be some validation on min/max value and allowance of 0 (for dynamic snat port allocation.)

Reply:
We've decided to include all fields for all phases of this project in 1 api version so we don't have to do multiple preview API releases to rollout the entirety of this feature. The phases are described in the PRD document linked on the parent Epic for the work item linked at the top of this PR.